buffer/src/main/java/io/netty/buffer/api/internal/Statics.java

```diff
@@ -175,10 +175,12 @@ private static ByteBuffer bbsliceFallback(ByteBuffer buffer, int fromOffset, int
             throw new IndexOutOfBoundsException(
                     "The limit of " + newLimit + " would be greater than capacity: " + buffer.capacity() + '.');
         }
+        final int pos = buffer.position();
+        final int lim = buffer.limit();
         try {
             return buffer.position(fromOffset).limit(newLimit).slice();
         } finally {
-            buffer.clear();
+            buffer.clear().limit(lim).position(pos);
```

Comment:
I think we should better do the followings otherwise the `lim` might be bigger hen the current position.
```suggestion
            buffer.clear().position(pos).limit(lim);
```